gammapy/spectrum/flux_point.py

```diff
@@ -1187,6 +1187,29 @@ def __init__(self, model, data, mask_fit=None, likelihood="chi2", mask_safe=None
                 " either 'chi2' or 'chi2assym'"
             )
 
+    def __repr__(self):
+        str_ = self.__class__.__name__
+        return str_
+
+    def __str__(self):
+        str_ = "Model       : {}\n".format(self.model)
+        str_ += "\n"
+        str_ += "Numbers  of flux points used for the fit  : {}\n".format(
+            len(np.where(self.mask)[0])
+        )
+        str_ += "Total Number of flux points : {} ".format(len(self.data.table))
+        str_ += "where {} were excluded for safe energy range ".format(
+            len(np.where(self.mask_safe == False)[0])
+        )
+        if self.mask_fit is not None:
+            str_ += "and {} were excluded by the user".format(
+                len(np.where(self.mask_fit == False)[0])
+            )
+        str_ += "\n"
+        str_ += "likelihood value   : {}\n".format(len(self._likelihood))
```

Comment:
Can this fail if no model has been set?
Should there be an if statement to check for the presence of a model before trying to print the likelihood value?